.github/workflows/chart-lint.yml

```diff
@@ -18,5 +18,20 @@ jobs:
 
       - uses: actions/setup-go@v3
 
+      - uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
       - name: Verify charts are upt-do-date
         run: make chart-lint
+
+  template:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
I suppose here it will not download again the files but will use the cache

Reply:
No, it will just checkout the git repo :shrug: 
The jobs `template` and `lint` (line 12) are separate, isolated and run in parallel, there's no dependency between.

Reply:
I noticed you added cache, but you restore only a key there

Reply:
I don't follow, what do you mean? The `lint` job can be optimized by caching the Go dependencies, since it runs checks that are based on Go. If there's no caching then it would simply download all the Go dependencies on-the-fly. The `template` job doesn't any caches, it only directly executes with `helm` which is preinstalled in the runner, no caches necessary.

Reply:
I'm going ahead and merge this. We can fix issues later on as well.